src/components/MonthSwitcher.vue

```diff
@@ -1,47 +1,50 @@
-<script setup>
-import { computed, reactive } from "vue";
-import { addMonths, subMonths, format } from "date-fns";
-import { mdiChevronLeft, mdiChevronRight } from "@mdi/js";
-const icons = reactive({
-  prev: mdiChevronLeft,
-  next: mdiChevronRight
-});
-const model = defineModel({ type: Date });
-function goPrev() {
-  model.value = subMonths(model.value, 1);
-}
-function goNext() {
-  model.value = addMonths(model.value, 1);
-}
+<script setup>  
+import { computed, reactive, onMounted } from "vue";  
+import { addMonths, subMonths, format } from "date-fns";  
+import { mdiChevronLeft, mdiChevronRight } from "@mdi/js";  
 
-const formattedDate = computed(() => {
-  return format(model.value, "MMMM yyyy");
-});
-</script>
+const icons = reactive({  
+  prev: mdiChevronLeft,  
+  next: mdiChevronRight  
+});  
 
-<template>
-  <div
-    style="
-      min-width: 300px;
-      min-height: 50px;
-      display: flex;
-      justify-content: center;
-    "
-  >
-    <v-btn
-      class="justify-center"
-      variant="text"
-      :icon="icons.prev"
-      @click="goPrev"
-    >
-    </v-btn>
-    <v-spacer />
-    <span style="align-self: center">
-      {{ formattedDate }}
-    </span>
-    <v-spacer />
-    <v-btn variant="text" :icon="icons.next" @click="goNext"></v-btn>
-  </div>
-</template>
+const model = reactive({ value: new Date() });  
```

Comment:
This does not work here. We need the model() functunality since it updates the Date in the ContentView.vue which again hands the updated value to the ReportingDisplay.vue compnent which uses it to fetch data.

Please also be carefull with reactive() in the context of Objects since it does not handle partial updates.